server/src/main/java/keywhiz/service/daos/SecretController.java

```diff
@@ -219,6 +224,23 @@ public SecretBuilder builder(String name, String secret, String creator, long ex
     return new SecretBuilder(transformer, secretDAO, name, encryptedSecret, hmac, creator, expiry);
   }
 
+  private void validateSecretSize(String base64EncodedSecret) {
+    if (config.getMaximumSecretSizeInBytesInclusive() == null) {
+      return;
+    }
+
+    int rawSecretLength = Base64.getDecoder().decode(base64EncodedSecret).length;
+
+    if (rawSecretLength > config.getMaximumSecretSizeInBytesInclusive()) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Secret is too large. Secret size %,d bytes exceeds maximum size of %,d bytes.",
+              rawSecretLength,
+              config.getMaximumSecretSizeInBytesInclusive()));
+    }
```

Comment:
New and improved:

* Configuration option name is now even more self-describing via the addition of "Inclusive"
* Exact raw secret size is now used for calculation (previously we were using the post-encryption, post-encoding version). This should be more user-friendly.
* Large byte values are now formatted with a thousands separator (this is what `%,d` means)